File: problems/problems_3618/solution.go
package problem3618

import (
	"encoding/json"
	"log"
	"strings"
)

var FLAG []bool

func init() {
	N := int(1e5)
	FLAG = make([]bool, N+1)
	FLAG[0] = true
	FLAG[1] = true
	for i := 2; i*i <= N; i++ {
		if !FLAG[i] {
			for j := i * 2; j <= N; j += i {
				FLAG[j] = true
			}
		}
	}
}

func splitArray(nums []int) (ans int64) {
	for i, num := range nums {
		if FLAG[i] {
			ans += int64(num)
		} else {
			ans -= int64(num)
		}
	}
	if ans < 0 {
		ans *= -1
	}
	return
}

func Solve(inputJsonValues string) any {
	inputValues := strings.Split(inputJsonValues, "\n")
	var nums []int

	if err := json.Unmarshal([]byte(inputValues[0]), &nums); err != nil {
		log.Fatal(err)
	}

	return splitArray(nums)
}

File: problems/problems_3618/solution.py
import solution
from typing import *


class Solution(solution.Solution):
    def solve(self, test_input=None):
        return self.splitArray(test_input)

    def splitArray(self, nums: List[int]) -> int:
        ans = 0
        for i, num in enumerate(nums):
            ans += num if PRIMES[i] else -num
        return abs(ans)

def primes(n):
    n = int(n)
    is_prime = [True] * (n + 1)
    is_prime[0] = is_prime[1] = False  # 0和1不是质数
    p = 2
    while p * p <= n:
        if is_prime[p]:
            for i in range(p * p, n + 1, p):
                is_prime[i] = False
        p += 1
    return is_prime

PRIMES = primes(1e5)
